tests/difference/core/quint_model/ccv.qnt

```diff
@@ -0,0 +1,619 @@
+        sentVSCPackets' = sentVSCPackets.keys().mapBy(
+            (chain) => 
+            sentVSCPackets.get(chain).union(newOutstandingPackets.get(chain).toSet())
+        )
+
+    
+    // the main step action 
+    action step: bool = any {
+        AdvanceTime,
+        nondet node = oneOf(Nodes)
+        nondet amount = oneOf(1.to(10))
+        votingPowerChange(node, amount),
+        recvPacketOnProvider(oneOf(ConsumerChains)),
+        recvPacketOnConsumer(oneOf(ConsumerChains)),
+        nondet chain = oneOf(chains)
+        endAndBeginBlock(chain),
+        val shouldAdvance = ConsumerChains.mapBy(
+            chain => 
+            nondet should = oneOf(Set(true, false))
+            should
+        )
+        AdvanceConsumers(shouldAdvance),
+    }
+
+    pure val nodePowerSet = Nodes.powerset()
+
+    def getArbitraryValidatorSet(): ValidatorSet = 
+        nondet numValidators = oneOf(1.to(Nodes.size()))
+        // toList has nondeterministic behaviour, so this gets arbitrary validators
+        nondet validators = oneOf(nodePowerSet.filter(s => s.size() == numValidators))
+        validators.mapBy(
+            validator => 
+            nondet votingPower = oneOf(1.to(10))
+            votingPower
+        )
+
+    // INITIALIZATION
+    action init: bool = 
+    all {
+        val validatorSets = chains.mapBy(
+            (chain) => 
+        // provider chain gets an arbitrary validator set, consumer chains have none
+            if (chain == ProviderChain) getArbitraryValidatorSet else Map()
+        )
+        all {
+            votingPowerHistories' = chains.mapBy(
+                (chain) => 
+                List(validatorSets.get(chain))
+            ),
+            runningValidatorSet' = validatorSets,
+        },
+        // each chain starts at time 0
+        curChainTimes' = chains.mapBy(
+            (chain) => 0
+        ),
+        // all consumer chains are unused
+        consumerStatus' = chains.mapBy(chain => UNUSED),
+        // no packets are outstanding
+        outstandingPacketsToProvider' = chains.mapBy(chain => List()),
+        outstandingPacketsToConsumer' = chains.mapBy(chain => List()),
+        // no maturations have been received by provider
+        receivedMaturations' = Set(),
+        // no packets have been sent to consumers
+        sentVSCPackets' = chains.mapBy(chain => Set()),
+        // no packets have been received by consumers, so no maturation times set
+        maturationTimes' = chains.mapBy(chain => Map()),
+        // validator set was not changed yet
+        providerValidatorSetChangedInThisBlock' = false
+    }
+
+    // PROPERTIES
+
+    // Every validator set on any consumer chain MUST either be or
+    // have been a validator set on the provider chain.
+    val ValidatorSetReplication: bool = 
+        chains.forall(
+            chain => chain.getCurrentValidatorSet().wasValidatorSetOnProvider()
+        )
+
+    // TESTS
```

Comment:
Do tests over the model always have to be explicitly defined? My understanding is that quint somehow enables fuzzed or implicitly generated test cases

Reply:
They do not need to be defined, Quint allows doing PBT in-built (checking that given invariants always hold).
This manual test case is mostly there for my benefit while writing the model to gradually check things I introduced, and can complement random testing.

Reply:
Nice!